sdk/eas/src/eas.ts

```diff
@@ -1,25 +1,85 @@
-export * from "./schema.js";
+import { SchemaRegistry } from "@ethereum-attestation-service/eas-sdk";
+import { validate } from "@settlemint/sdk-utils/validation";
+import { JsonRpcProvider, Wallet } from "ethers";
+import { type ClientOptions, ClientOptionsSchema } from "./client-options.schema.js";
+import type { RegisterSchemaOptions } from "./types.js";
+import { buildSchemaString, validateSchemaFields } from "./validation.js";
 
-export function createEASClient() {
-  return {
-    submitAttestation() {
-      console.log("[EAS] Submitting attestation");
-    },
+/**
+ * Creates an EAS client for interacting with the Ethereum Attestation Service.
+ *
+ * @param options - Configuration options for the client
+ * @returns An object containing the EAS client instance
+ * @throws Will throw an error if the options fail validation
+ *
+ * @example
+ * ```ts
+ * import { createEASClient } from '@settlemint/sdk-eas';
+ * import { Wallet } from 'ethers';
+ *
+ * // Using a private key
+ * const client = createEASClient({
+ *   schemaRegistryAddress: "0x1234567890123456789012345678901234567890",
+ *   attestationAddress: "0x1234567890123456789012345678901234567890",
+ *   blockchainNode: "http://localhost:8545",
+ *   wallet: "your-private-key-here"
+ * });
+ *
+ * // Or using a Wallet instance
+ * const wallet = new Wallet("your-private-key-here");
+ * const client = createEASClient({
+ *   schemaRegistryAddress: "0x1234567890123456789012345678901234567890",
+ *   attestationAddress: "0x1234567890123456789012345678901234567890",
+ *   blockchainNode: "http://localhost:8545",
+ *   wallet
+ * });
+ * ```
+ */
+export function createEASClient(options: ClientOptions) {
```

Comment:
Document ClientOptions, RegisterSchemaOptions and re-export it so it can be used by implementors